src/dataframe/dataframe.jl

```diff
@@ -107,11 +109,21 @@ struct DataFrame <: AbstractDataFrame
         minlen, maxlen = extrema(lengths)
         if minlen != maxlen || minlen == maxlen == 1
             # recycle scalars
-            for i in 1:length(columns)
-                isa(columns[i], AbstractArray) && continue
-                columns[i] = fill!(Tables.allocatecolumn(typeof(columns[i]), maxlen),
-                                   columns[i])
-                lengths[i] = maxlen
+            for (i, col) in enumerate(columns)
+                if col isa AbstractArray
+                    if ndims(col) == 0
+                        x = col[]
```

Comment:
You could probably avoid repeating the code by doing `if col isa Ref || (col isa AbstractArray and ndims(col) == 0)` and then `elseif !(col isa AbstractArray)`.

Reply:
also fixed